### PR TITLE
Ignore recursively

### PR DESCRIPTION
**Pull request by jgershen**
first, add a test for our old ignore-key functionality.
then add the ability to ignore keys recursively while testing an integration (with a test for that as well).

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -169,10 +169,18 @@ Assertion.prototype.maps = function(fixture, settings, options){
...
 
   if (options.ignored) {
-    options.ignored.map(function(name) {
-      delete actual[name]
-      assert(!expected[name], "ignored expected value:" + name);
-    });
+    // Recursively strip all keys given in options.ignored from actual, and from
+    // all objects contained in or nested under actual.
+    var stripIgnoredFromResult = function(obj) {
+      for(prop in obj) {
```
Comment (thehydroimpulse) [inline lib/index.js]: Nitpicky, but `for`, `if` should probably have a space between the parenthesis. I don't think we're linting in these repos

File: lib/index.js [hunk 2]
```diff
@@ -169,10 +169,18 @@ Assertion.prototype.maps = function(fixture, settings, options){
...
+        if(options.ignored.indexOf(prop) != -1) {
+          delete obj[prop];
+        } else if (typeof obj[prop] === 'object') { // recurse on contained objs
```
Comment (thehydroimpulse) [inline lib/index.js]: You might want to check if it's an array too. Arrays are also objects and `typeof [] === 'object'` is `true`.

```js
else if (typeof obj[prop] === 'object' && !(obj[prop] instanceof Array)) {
  // ...
}
```

File: lib/index.js (hunk 1 above)
Comment (jgershen) [inline lib/index.js]: nitpicking is welcome! good catch

File: lib/index.js (hunk 2 above)
Comment (jgershen) [inline lib/index.js]: actually we want to recurse on arrays. I need to fix the delete statement though, in case we have recursed down to an array.

Comment (thehydroimpulse) [inline lib/index.js]: Gotcha! Sounds good to me!

**Comment by thehydroimpulse** [comment]
@jgershen This PR LGTM

**Comment by jgershen** [comment]
fixed to handle arrays

**Pull request merged**